docx/oxml/text/run.py

```diff
@@ -90,14 +90,25 @@ def text(self):
         equivalent.
         """
         text = ''
-        for child in self:
-            if child.tag == qn('w:t'):
-                t_text = child.text
-                text += t_text if t_text is not None else ''
-            elif child.tag == qn('w:tab'):
-                text += '\t'
-            elif child.tag in (qn('w:br'), qn('w:cr')):
-                text += '\n'
+        if CT_FldChar.numOfNestedFldChar > 0:
+            for child in self:
+                if child.tag == qn('w:fldChar'):
+                    if child.fldCharType == 'begin':
+                        CT_FldChar.numOfNestedFldChar += 1
+                    else:
+                        CT_FldChar.numOfNestedFldChar -= 1
```

Comment:
Do we need to check if the type is `end` here?
If we encounter `separate` this will start tracking data from that point since the counter will be decremented. Am I thinking correctly on this one?

Reply:
If `w:fldChar` has a 'separator' type in it, it means that form the 'separator' through the 'end' the text is shown.
So we only hide text between 'begin' and 'end', or between 'begin' and 'separate'.
You can see this behaviour in complex field for date, between 'begin' and 'separate' is instruction text that determents how to format date. Between  'separate' and 'end' is the actual date value in the correct format that the word document shows.

Reply:
Makes sense, thanks for clarifying. It would be nice if you could include part of this explanation in the code. I can see this being a source of confusion in the future.